presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoResultSet.java

```diff
@@ -1824,7 +1899,7 @@ private static boolean isQuery(StatementClient client)
 
         private void checkInterruption(Throwable t)
         {
-            if (Thread.currentThread().isInterrupted()) {
+            if (Thread.currentThread().isInterrupted() || parent.isInterrupted()) {
```

Comment:
I don't think we need this check here. The purpose of this was to cleanly exit when current thread (which did all the work before this multi-threading change) was interrupted. Instead, we should call `client.close()` if we get `InterruptedException` inside `AsyncIterator`.

Reply:
Move this check to `AsyncIterator`.
This is needed for `TestPrestoDriver.testQueryCancelByInterrupt`